ui/app/src/components/Header.tsx

```diff
@@ -11,17 +11,99 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-import { AppBar, Switch, Toolbar, Typography } from '@material-ui/core';
+import {
+  AppBar,
+  Box,
+  Button,
+  CircularProgress,
+  Divider,
+  Menu,
+  MenuItem,
+  Switch,
+  Theme,
+  Toolbar,
+  Typography,
+} from '@mui/material';
+import { ChevronDown } from 'mdi-material-ui';
+import { SxProps } from '@mui/system/styleFunctionSx/styleFunctionSx';
+import { MouseEvent, useState } from 'react';
+import { ProjectModel, useProjectQuery } from '../model/project-client';
+import Toast from './Toast';
+
+function ProjectMenu(props: { projectList: ProjectModel[] }): JSX.Element {
+  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
+
+  const handleMenu = (event: MouseEvent<HTMLElement>) => {
+    setAnchorEl(event.currentTarget);
+  };
+  const handleCloseMenu = () => {
+    setAnchorEl(null);
+  };
+  return (
+    <>
+      <Button
+        aria-label="list of the available project"
+        aria-controls="menu-project-list-appbar"
+        aria-haspopup="true"
+        color="inherit"
+        endIcon={<ChevronDown />}
+        onClick={handleMenu}
+      >
+        Projects
+      </Button>
+      <Menu
+        id="menu-project-list-appbar"
+        anchorEl={anchorEl}
+        anchorOrigin={{
+          vertical: 'bottom',
+          horizontal: 'left',
+        }}
+        keepMounted
+        open={Boolean(anchorEl)}
+        onClose={handleCloseMenu}
+      >
+        {props.projectList.map((value, index) => {
+          return (
+            // TODO when routing is in place, use the button to redirect to the project page
+            <MenuItem key={index} onClick={handleCloseMenu}>
+              {value.metadata.name}
+            </MenuItem>
+          );
+        })}
+      </Menu>
+    </>
+  );
+}
+
+const style: SxProps<Theme> = {
+  display: 'flex',
+  flexDirection: 'row',
+};
 
 export default function Header(): JSX.Element {
+  const { response, loading, error } = useProjectQuery();
   return (
     <AppBar position="relative">
       <Toolbar>
-        <Typography variant="h6" sx={{ flexGrow: 1 }}>
-          Perses
-        </Typography>
+        <Box sx={style} flexGrow={1}>
+          <Typography variant="h6" sx={{ marginRight: '1rem' }}>
+            Perses
+          </Typography>
+          <Divider
+            orientation="vertical"
+            flexItem
+            sx={{ borderRightColor: 'rgba(255,255,255,0.2)' }}
+          />
+          {loading ? (
+            <CircularProgress size="1rem" />
+          ) : (
+            response && <ProjectMenu projectList={response} />
+          )}
+        </Box>
         <Switch />
       </Toolbar>
+      {/* TODO manage in a single place the pull of fetching error */}
```

Comment:
What do you mean with "pull" in "pull of fetching error"? Pool as in multiple errors?

Reply:
ah yeah `pool of error` as it can comes from multiple different sources. It won't scale if each component has to integrate the way to display the error

Reply:
Yep, makes sense!